sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -94,6 +94,10 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
       Iterator[UnsafeRowPair] = {
       map.prefixScan(prefixKey)
     }
+
+    override def valuesIterator(key: UnsafeRow, colFamilyName: String): Iterator[UnsafeRow] = {
+      Seq(get(key, colFamilyName)).iterator
```

Comment:
Where is this used ?

Reply:
This used to read the list values (encoded in a single key) for ListState. More specifically, its used in ListStateImpl.get() function. (line 45 in ListStateImpl.scala).